Lib/test/support/__init__.py

```diff
@@ -1077,7 +1077,7 @@ def run_unittest(*classes):
         elif isinstance(cls, valid_types):
             suite.addTest(cls)
         else:
-            suite.addTest(unittest.makeSuite(cls))
+            suite.addTest(unittest.TestLoader().loadTestsFromTestCase(cls))
```

Comment:
There is also a use of `findTestCases()` above.

BTW, could not `TestLoader()` be shared?